msprime/trees.py

```diff
@@ -650,6 +676,12 @@ def draw(
         :param bool show_mutation_labels: If True, show labels for mutations.
         :param bool show_internal_node_labels: If True, show labels for internal nodes.
         :param bool show_leaf_node_labels: If True, show labels for leaf nodes.
+        :param list branch_colours: A list giving colours for edges in the tree.
+            If this is an array such as ['red','blue'], the indexes of the array
+            correspond to the node whose edges are coloured. If a dict, the keys
+            give the node numbers for each colour. Unspecified edges are coloured  
```

Comment:
> The nodes iterator returns all nodes in the subtree u in preorder. So, all branches in the subtree will be coloured. Try it out.

Oh, I see. Thanks.

> You can colour them using precisely the same algorithm in client code.

Had just noticed. Sorry.

> We can make a cookbook of clever ways to colour trees in the documentation

Yes, as long as we can remember how to do it, that seems fine.